File: benchmarks/nyc_taxi/python/q19_core_mutate.py
import argparse
import time

import pandas as pd


def main() -> None:
    parser = argparse.ArgumentParser()
    parser.add_argument("--input-parquet", required=True, help="Path to a materialized parquet benchmark input")
    args = parser.parse_args()

    started = time.perf_counter()
    frame = pd.read_parquet(args.input_parquet)
    positive_total = frame["total_amount"] > 0
    result = frame.loc[positive_total].assign(
        tip_percent=frame.loc[positive_total, "tip_amount"] / frame.loc[positive_total, "total_amount"] * 100.0
    )

    print(result.head(10))
    print(f"ROWS_SCANNED={len(frame.index)}")
    print(f"ROWS_RETURNED={len(result.index)}")
    print(f"ELAPSED_SEC={time.perf_counter() - started:.6f}")


if __name__ == "__main__":
    main()

File: benchmarks/nyc_taxi/r/q19_core_mutate.R
args <- commandArgs(trailingOnly = TRUE)
if (length(args) < 1) {
  stop("usage: Rscript q19_core_mutate.R <materialized-parquet-path>", call. = FALSE)
}

suppressPackageStartupMessages({
  library(arrow)
  library(dplyr)
})

started <- Sys.time()
input_path <- args[[1]]
frame <- read_parquet(input_path, as_data_frame = TRUE)

result <- frame |>
  filter(total_amount > 0) |>
  mutate(tip_percent = tip_amount / total_amount * 100)

print(head(result, 10))
cat(sprintf("ROWS_SCANNED=%s\n", nrow(frame)))
cat(sprintf("ROWS_RETURNED=%s\n", nrow(result)))
cat(sprintf("ELAPSED_SEC=%.6f\n", as.numeric(difftime(Sys.time(), started, units = "secs"))))
